src/frontend/src/stream_fragmenter/mod.rs

```diff
@@ -229,8 +229,13 @@ fn build_fragment(
 ) -> Result<StreamNode> {
     // Update current fragment based on the node we're visiting.
     match stream_node.get_node_body()? {
-        NodeBody::Source(_) => {
+        NodeBody::Source(src) => {
             current_fragment.fragment_type_mask |= FragmentTypeFlag::Source as u32;
+            // Note: For creating table with connector, the source id is left with zero and should
+            // not be recorded in dependent table ids.
+            if let Some(inner) = &src.source_inner && inner.source_id != 0 {
```

Comment:
What about extracting a `PLACEHOLDER` constant so that it'll be more explicit?

https://github.com/risingwavelabs/risingwave/blob/896d92ee3cad6095e021f74b0686777f3d5a33c6/src/frontend/src/handler/create_table.rs#L384